### PR TITLE
js/bundle: Add command line tool

### PR DESCRIPTION
**Pull request by irori**
This adds `gen-webbundle` CLI tool to the `wbn` package, which enables JS developers to build a web bundle from a local directory, without having to install the Go tools.

Internally it uses `addFilesRecursively`, a new high-level method of `BundleBuilder`.

### CODE REVIEW

File: js/bundle/src/encoder.ts [hunk 1]
```diff
@@ -66,6 +69,35 @@ export class BundleBuilder {
...
+    for (const file of files) {
+      const filePath = path.join(dir, file);
+      if (fs.statSync(filePath).isDirectory()) {
+        this.addFilesRecursively(filePath, baseURL + file + '/');
+      } else if (file === 'index.html') {
+        // If the file name is 'index.html', create an entry for baseURL itself
+        // and another entry for baseURL/index.html which redirects to baseURL.
+        // This matches the behavior of gen-bundle.
+        this.addFile(filePath, baseURL);
+        this.addExchange(baseURL + file, 301, { Location: './' }, '');
```
Comment (sisidovski) [inline js/bundle/src/encoder.ts]: My understanding might be wrong though, if there is a file located at `/foo/bar/baz/index.html`, this code looks redirecting to `baseURL/index.html`. Is that a correct behavior?

File: js/bundle/package.json [hunk 2]
```diff
@@ -9,6 +9,9 @@
     "build": "tsc",
     "test": "jasmine tests/*.js"
   },
+  "bin": {
+    "gen-webbundle": "./bin/gen-webbundle"
```
Comment (sisidovski) [inline js/bundle/package.json]: IMO: Just using package name (wbn) as a command name does more make sense for me. Why do you make this name?

File: js/bundle/src/encoder.ts (hunk 1 above)
Comment (irori) [reply js/bundle/src/encoder.ts]: Note that the recursive call at line 88 updates `baseURL` adding the directory name.

Here, `baseURL`'s value should be `<user-specified-baseURL>/foo/bar/baz/`, so this exchange would redirect `<user-specified-baseURL>/foo/bar/baz/index.html` to `<user-specified-baseURL>/foo/bar/baz/`.

File: js/bundle/package.json (hunk 2 above)
Comment (irori) [reply js/bundle/package.json]: Yeah, that makes sense to me. Renamed the command.

We may add other CLI features in the future (such as `dump-bundle` equivalent), but then we could switch to the subcommand style (e.g. `wbn gen ...` / `wbn dump ...`).

**Comment by irori** [comment]
@sisidovski Thank you for the review!
@kinu Would you approve this PR?

File: js/bundle/src/encoder.ts (hunk 1 above)
Comment (sisidovski) [reply js/bundle/src/encoder.ts]: Ah! I was missing it! Thank you for the comment!

**Comment by irori** [comment]
Tweaked a bit: changed the argument order of `addFiles(Recursively)` to take URL as first argument, for consistency with `addExchange()`.

**Pull request merged**